nifi-nar-bundles/nifi-smb-bundle/nifi-smb-client-provider-api/src/main/java/org/apache/nifi/services/smb/NiFiSmbClient.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.nifi.services.smb;
+
+import java.io.OutputStream;
+import java.util.stream.Stream;
+
+public interface NiFiSmbClient {
```

Comment:
Although the `NiFi` prefix helps to distinguish this interface from the SMBJ `SmbClient`, prefixing components with `NiFi` is not a great pattern because the package name already indicates NiFi association. What about naming this `SmbClientService`?
```suggestion
public interface SmbClientService {
```